src/vs/workbench/contrib/terminalContrib/suggest/common/terminalSuggestConfiguration.ts

```diff
@@ -195,4 +191,59 @@ export const terminalSuggestConfiguration: IStringDictionary<IConfigurationPrope
 
 };
 
+let terminalSuggestProvidersConfiguration: IConfigurationNode | undefined;
+
+export function registerTerminalSuggestProvidersConfiguration(availableProviders?: string[]) {
+	const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Configuration);
+
+	const oldProvidersConfiguration = terminalSuggestProvidersConfiguration;
+
+	// Create properties for the providers setting dynamically
+	const providersProperties: IStringDictionary<IConfigurationPropertySchema> = {};
+	const defaultValue: IStringDictionary<boolean> = {
+		// Always include known built-in providers as defaults even if not yet registered
+		'terminal-suggest': true,
+		'builtinPwsh': false,
```

Comment:
With https://github.com/microsoft/vscode/issues/254872 we no longer need to specify any defaults. Should we merge that one first?